privacyidea/lib/tokens/webauthntoken.py

```diff
@@ -1181,7 +1181,23 @@ def check_otp(self, otpval, counter=None, window=None, options=None):
                 raise ValueError("When performing WebAuthn authorization, options must contain user")
 
             uv_req = get_optional(options, FIDO2PolicyAction.USER_VERIFICATION_REQUIREMENT)
-            challenge = binascii.unhexlify(get_required(options, "challenge"))
+            # Check if challenge is base64 encoded to be able to use login via passkey with webauthn
+            challenge = get_required(options, "challenge")
+            challenge: str = challenge.replace("=", "")
+            challenge_decoded = None
+            if len(challenge) % 2 == 0:
+                try:
+                    challenge_decoded = webauthn_b64_encode(binascii.unhexlify(challenge))
+                except Exception as ex:
+                    log.warning(f"Challenge {get_required(options, 'challenge')} is not hex encoded. {ex}. "
+                                f"Attempting to decode it as base64url.")
+            if not challenge_decoded:
+                try:
+                    challenge_decoded = bytes_to_base64url(challenge.encode("utf-8"))
```

Comment:
so the webauthn token **can** work with base64url encoded data?

Reply:
yes it can work with challenges generated with /validate/initialize, as shown in the tests